build.gradle

```diff
@@ -96,9 +101,17 @@ task sourcesJar(type: Jar) {
     	configurations.extraLibs.collect { it.isDirectory() ? it : zipTree(it) }
     }
 }
+task uberJar(type: Jar) {
```

Comment:
Same as before - Is this `manifest`block of code necessary to the library of was it only necessary for local tests? If the latter is true, it must be removed from the final version.

Reply:
Also a convenience, just like javadocJar and sourcesJar. This provides a version of the library with everything included. This can be useful for some users. It complements the standard Jar which is also generated.

Reply:
Same as before -  CST is not an application, it is a library. Differently from TROCA, it will never be run as a Java JAR application, but only be imported into applications from the maven repository. This is not needed.

Reply:
André, I am sorry to disagree with you here. This is more than just a convenience that MIGHT be useful ... in fact, it might be EXTREMELY USEFUL in some specific situations like actual TROCA. Because TROCA now must be run from inside a docker container, TROCA requires a uberJar to be deployed (or, at least while it is under development, this makes development much easier). In a specific scenario where I am working right now, it is not possible to create a uberJar for TROCA, containing both CST and its dependencies, plus MECA and its dependencies. Gradle fails in providing that (unfortunately, gradle fails to provide dependency transitivity while using local files for the libraries). The only way I was able to make it to work was building both CST and MECA as uberJars. I believe that, even though I agree that it is not a common scenario, and in most of the situations this might be not required, there might be situations, like that, where the possibility of having a uberJar might be of great use for the library user. These are situations where the dependency transitivity fails, and there are specific situations where gradle fails in providing dependency transitivity (particularly while using local files for libraries). Why make things more difficult to a potential user (and particularly when the users are ourselves, and we are suffering due to that) ? To save the space of 3 lines of code in build.gradle ? I believe that if we can provide something which is of benefit to a potential user, this should be included. It doesn't harm. If the user does not want to use the uberJar it will not affect other uses. It is not forcing the built jar library to be a uberJar, but providing a second jar library with this convenience, in the case it might be useful (and our current case shows that this might be the case in some situations). The practical effect is the same as generating the javadoc and sourceZip .... a library user might not be interested into that also. But it is pretty convenient, if it can have it ready while they are necessary.